src/Telemetry/TelemetryService.php

```diff
@@ -179,10 +183,23 @@ public function reportUsageData(): int|bool
      *
      * @param mixed $event_data The event data to set.
      */
-    public function trackApiRequestEvent(mixed $event_data): void
+    public function trackApiRequestEvent(array $event_data): void
     {
         if (!empty($this->isTelemetryEnabled())) {
             $this->reportClickEvent($event_data);
         }
     }
+
+    private function normalizeUrl(string $url): string
+    {
+        $parsed = parse_url($url);
+        $path = $parsed['path'] ?? '';
+        $fragment = isset($parsed['fragment']) ? '#' . $parsed['fragment'] : '';
+        if (!empty($GLOBALS['webroot'])) {
+            $normalized = preg_replace('#^(' . $GLOBALS['webroot'] . ')?#', '', $path);
+        } else {
+            $normalized = $path;
+        }
+        return ($normalized . $fragment);
+    }
```

Comment:
@sjpadgett , this is working nicely and basically look like urls that core is bringing in now. only doing this on the portal urls. 
Thank goodness had mechanism to test webroot via easy dev:
```
webroot-management:
  cwb, change-webroot-blank          Change webroot to be blank (this is default setting of environment)
  cwo, change-webroot-openemr        Change webroot to be openemr
```

Reply:
yep empty webroot is an issue that bites us all the time for those that alias elsewhere. I try to remember to test on localhost but still get caught. Nice catch!